### ISSUE TITLE
Example for XRay as a sidecar

### ISSUE THREAD

**Issue by iamhopaul123** [aws/copilot-cli#2931]
As a user, I would like to use AWS [X-Ray](https://aws.amazon.com/xray/) to analyze and debug my distributed applications. More specifically, it could be run as a sidecar container. It would be great if Copilot can provide an example on this use case.

**Comment by iamhopaul123** [comment]
## Example
The example consists of a publisher service that keeps publishing to a SQS queue constantly, and a worker service consuming from a queue that subscribes to the topic, and getting scaled based on the queue.

https://github.com/iamhopaul123/copilot-xray-example

## Deploy Steps
The commands I ran:
```
1. copilot app init -n loadtest
2. copilot env init -n test
// Create the publisher.
3. copilot svc init -n loadtester
4. copilot svc deploy -n loadtester
// Create the worker service.
5. copilot svc init -n worker
6. copilot svc deploy -n worker
```

## X-Ray Console
<img width="911" alt="Screen Shot 2021-10-18 at 4 42 02 PM" src="https://user-images.githubusercontent.com/51377557/137820934-3a89d0c1-1a27-480f-9987-b90f212486cf.png">

**Comment by fcirone** [comment]
Can we have a similar topic for an app mesh as a sidecar? Thanks!

**Comment by efekarakus** [comment]
Hi @fcirone 👋 , thanks for the feedback! Would you mind expanding on your usecases? 
Are there particular app mesh features you'd like to use? Is there anything you see copilot help simplify for the app mesh integration?

**Comment by fcirone** [comment]
Hi @efekarakus, I am creating an app using microservices and aws copilot to provide all the infrastructure.
I didnt find a way to add App Mesh as a sidecar container using copilot. 

Is there a way to add App Mesh as a sidecar like xray sidecar in the example?

**Comment by corey-cole** [comment]
I had to create a PoC for both X-Ray and OTEL, and once I found the secret, it was surprisingly easy.


```
sidecars:
  xray_sidecar:
    image: 'public.ecr.aws/xray/aws-xray-daemon:3.x'
    port: 2000/udp
```

OTEL was a little trickier, in that you have to store your OTEL config within an SSM parameter and then pass it to the collector as an envvar:

```
# Can provide configuration via AOT_CONFIG_CONTENT
# Can also provide 'cmd' but then we have to map a file into the sidecar
sidecars:
  otel_sidecar:
    image: 'public.ecr.aws/aws-observability/aws-otel-collector:latest'
    secrets:
      AOT_CONFIG_CONTENT: /copilot/${COPILOT_APPLICATION_NAME}/${COPILOT_ENVIRONMENT_NAME}/secrets/otel_config

```

OTEL has the benefit of being able to ship both X-Ray and publish internal metrics to something like New Relic.

```
# This is the config stored in SSM.  If it gets too big, use an advanced parameter
receivers:
  awsxray:
    transport: udp
  awsecscontainermetrics: # collect 52 metrics
  otlp:
    protocols:
      grpc:
        endpoint: 0.0.0.0:4317

processors:
  batch:

exporters:
  awsxray:
    region: us-west-2
  otlp:
    endpoint: otlp.nr-data.net:4317
    headers: 
      api-key: YOUR-API-KEY-HERE

service:
  pipelines:
    traces:
      receivers: [awsxray,otlp]
      processors: [batch]
      exporters: [awsxray]
    metrics:
      receivers: [awsecscontainermetrics ]
      exporters: [otlp]
```

**Comment by dannyrandall** [comment]
Closed with #3457.
